src/state/farms/hooks.ts

```diff
@@ -13,6 +13,18 @@ import { State, Farm, FarmsState } from '../types'
 export const usePollFarmsData = (includeArchive = false) => {
   const dispatch = useAppDispatch()
   const { slowRefresh } = useRefresh()
+
+  useEffect(() => {
+    const farmsToFetch = includeArchive ? farmsConfig : nonArchivedFarms
+    const pids = farmsToFetch.map((farmToFetch) => farmToFetch.pid)
+
+    dispatch(fetchFarmsPublicDataAsync(pids))
+  }, [includeArchive, dispatch, slowRefresh])
+}
+
+export const usePollFarmsWithUserData = (includeArchive = false) => {
+  const dispatch = useAppDispatch()
+  const { slowRefresh } = useRefresh()
```

Comment:
I don't understand how this reduces the number of calls. It does not fetch user data unless there is an account, and if there is an account then we want to fetch user data.

Reply:
In Pools page we dont need user farm data instead we only need public farm data for prices. Farms with user data hook will be used in Farms and the other one will be used used in Pools page.

Reply:
I think you're right Memoyil - we aren't interested in the users' approval state/amount staked/user balance/earnings for farms while on pools. This could be a fairly quick performance win.